test/SignAndGet.test.js

```
@@ -50,241 +50,311 @@ describe("SignAndGet", () => {
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        undefined,
+        false
+      );
+    });
+
+    it("sets the correct custom headers for OAuth Tokens when allData is false", () => {
+      expect(OAuth.OAuth).toBeCalledWith(
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        { Accept: "application/vnd.deere.axiom.v3+json", No_Paging: false }
+      );
+    });
   });
 
-  it("return an error when there is an http status code below 200 from OAuth Sign Request endpoint", async () => {
-    const statusCode = chance.natural({ min: 0, max: 199 });
-    jest.spyOn(OAuth, "OAuth").mockImplementation(() => ({
-      get: (link, accessToken, accessTokenSecret, callback) => {
-        callback(undefined, undefined, { statusCode });
-      },
-    }));
-    const expectedStatusText = chance.string();
-    getStatusText.mockImplementation(() => {
-      return expectedStatusText;
+  describe("OAuth get returns a http status code below 200", () => {
+    let httpStatusCode;
+    let expectedStatusText;
+    let result;
+
+    beforeEach(() => {
+      httpStatusCode = chance.natural({ min: 0, max: 199 });
+      expectedStatusText = chance.string();
+
+      jest.spyOn(OAuth, "OAuth").mockImplementation(() => ({
+        get: (link, accessToken, accessTokenSecret, callback) => {
+          callback(undefined, undefined, { statusCode: httpStatusCode });
+        },
+      }));
+
+      getStatusText.mockImplementation(() => {
+        return expectedStatusText;
+      });
+
+      result = underTest.doSignAndGet(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
     });
 
-    await expect(
-      underTest.doSignAndGet(chance.string(), chance.string(), chance.string())
-    ).resolves.toMatch(expectedStatusText);
+    it("return an error", async () => {
```

probably "returns an error", or even better "should return an error"